pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -46,7 +46,14 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 	defer glog.V(3).Infof("Leaving Allocate ...")
 
 	queues := util.NewPriorityQueue(ssn.QueueOrderFn)
+	// map[queueId]PriorityQueue(*api.JobInfo)
 	jobsMap := map[api.QueueID]*util.PriorityQueue{}
+	// map[queueId]PriorityQueue(namespaceName)
+	namespaceMap := map[api.QueueID]*util.PriorityQueue{}
+	// map[queueId]map[namespaceName]PriorityQueue(*api.JobInfo)
+	jobInNamespaceMap := map[api.QueueID]map[string]*util.PriorityQueue{}
+
+	namespaceOrderEnabled := ssn.NamespaceOrderEnabled()
```

Comment:
This feature should be enabled by default, it dependent on plugins to enabled/disable namespace order.

Reply:
This `namespaceOrderEnabled` depends on the length of `session.NamespaceOrderFns`. If no plugin register that Fn, I think the namespace order should be disabled.

Because if no plugin register the `NamespaceOrderFn`, the job order should have its affect among all namespace, but not just within namespace. That's why I use different strategy between enabled and disable case.

Futher more, in another case, the jobs in different namespace but with the same namespace order should also be judged equally, but not just choose one namespace and then choose the another one. I mark this case as todo in `session.NamespaceOrderFn` func.